### ISSUE TITLE
Does not work but no exception be thrown

### ISSUE THREAD

**Issue by coldsmirk** [fury-project/dubbo-serialization-fury#2]
### HelloService.java

```java
public interface HelloService {

    String sayHello(String name);
}
```

### HelloServiceImpl

```java
public class HelloServiceImpl implements HelloService {

    @Override
    public String sayHello(String name) {
        System.out.println("hello " + name);

        return "hello " + name + " from dubbo";
    }
}
```

### Provider.java

```java
public class Provider {
    public static void main(String[] args) {
        var sc = new ServiceConfig<HelloService>();
        sc.setInterface(HelloService.class);
        sc.setRef(new HelloServiceImpl());

        var rc = new RegistryConfig("zookeeper://127.0.0.1:2181");
        // rc.setTimeout(250000);
        //    rc.setParameters(Map.of(
        //        "blockUntilConnectedWait", "250000"
        //    ));

        var protocolConfig = new ProtocolConfig(CommonConstants.DUBBO, 10880);
        protocolConfig.setSerialization("fury");
        DubboBootstrap.getInstance()
            .application(new ApplicationConfig("provider-demo"))
            .service(sc)
            .registry(rc)
            .protocol(protocolConfig)
            .start()
            .await();
    }
}
```

## Consumer.java

```java
public class Consumer {

    public static void main(String[] args) {
        var referenceConfig = new ReferenceConfig<HelloService>();
        referenceConfig.setInterface(HelloService.class);

        var protocolConfig = new ProtocolConfig(CommonConstants.DUBBO, 10880);
        DubboBootstrap.getInstance()
            .reference(referenceConfig)
            .application(new ApplicationConfig("consumer"))
            .registry(new RegistryConfig("zookeeper://127.0.0.1:2181"))
            .protocol(protocolConfig)
            .start(true);

        System.out.println(referenceConfig.get().sayHello("venus"));
    }
}
```

## Gradle deps

```gradle
dependencies {
    testImplementation platform('org.junit:junit-bom:5.9.1')
    testImplementation 'org.junit.jupiter:junit-jupiter'
    // https://mvnrepository.com/artifact/org.apache.dubbo/dubbo
    implementation 'org.apache.dubbo:dubbo:3.2.5'
    // https://mvnrepository.com/artifact/org.apache.dubbo/dubbo-dependencies-zookeeper
    implementation 'org.apache.dubbo:dubbo-dependencies-zookeeper:3.2.5'
    implementation 'org.furyio:fury-core:0.1.0'
    implementation 'org.furyio:dubbo-serialization-fury:0.2.0-SNAPSHOT'

}
```

**Note: `org.furyio:dubbo-serialization-fury:0.2.0-SNAPSHOT` is my local installed jar**

## Zookeeper env

```shell
docker run -d --name zookeeper -p 2181:2181 --restart=unless-stopped zookeeper:3.8
```

## Result

```log
log4j:WARN No appenders could be found for logger (org.apache.dubbo.rpc.model.FrameworkModel).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Exception in thread "main" org.apache.dubbo.rpc.RpcException: Failed to invoke the method sayHello in the service com.ilxqx.dubbo.demo.HelloService. Tried 3 times of the providers [192.168.20.1:10880] (1/1) from the registry 127.0.0.1:2181 on the consumer 192.168.20.1 using the dubbo version 3.2.5. Last error is: Invoke remote method timeout. method: sayHello, provider: dubbo://192.168.20.1:10880/com.ilxqx.dubbo.demo.HelloService?application=provider-demo&background=false&category=providers,configurators,routers&check=false&deprecated=false&dubbo=2.0.2&dynamic=true&executor-management-mode=isolation&file-cache=true&generic=false&interface=com.ilxqx.dubbo.demo.HelloService&methods=sayHello&pid=92976&prefer.serialization=fury&release=3.2.5&serialization=fury&service-name-mapping=true&side=provider&sticky=false&unloadClusterRelated=false, cause: Timeout after 1000ms waiting for result.
	at org.apache.dubbo.rpc.cluster.support.FailoverClusterInvoker.doInvoke(FailoverClusterInvoker.java:114)
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invoke(AbstractClusterInvoker.java:344)
	at org.apache.dubbo.rpc.cluster.router.RouterSnapshotFilter.invoke(RouterSnapshotFilter.java:46)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.monitor.support.MonitorFilter.invoke(MonitorFilter.java:108)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.rpc.cluster.filter.support.MetricsClusterFilter.invoke(MetricsClusterFilter.java:56)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.rpc.protocol.dubbo.filter.FutureFilter.invoke(FutureFilter.java:52)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.spring.security.filter.ContextHolderParametersSelectedTransferFilter.invoke(ContextHolderParametersSelectedTransferFilter.java:41)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.metrics.filter.MetricsFilter.invoke(MetricsFilter.java:73)
	at org.apache.dubbo.rpc.cluster.filter.support.MetricsConsumerFilter.invoke(MetricsConsumerFilter.java:37)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.rpc.cluster.filter.support.ConsumerClassLoaderFilter.invoke(ConsumerClassLoaderFilter.java:40)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.rpc.cluster.filter.support.ConsumerContextFilter.invoke(ConsumerContextFilter.java:116)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CallbackRegistrationInvoker.invoke(FilterChainBuilder.java:196)
	at org.apache.dubbo.rpc.cluster.support.wrapper.AbstractCluster$ClusterFilterInvoker.invoke(AbstractCluster.java:91)
	at org.apache.dubbo.rpc.cluster.support.wrapper.MockClusterInvoker.invoke(MockClusterInvoker.java:103)
	at org.apache.dubbo.rpc.cluster.support.wrapper.ScopeClusterInvoker.invoke(ScopeClusterInvoker.java:169)
	at org.apache.dubbo.registry.client.migration.MigrationInvoker.invoke(MigrationInvoker.java:284)
	at org.apache.dubbo.rpc.proxy.InvocationUtil.invoke(InvocationUtil.java:61)
	at org.apache.dubbo.rpc.proxy.InvokerInvocationHandler.invoke(InvokerInvocationHandler.java:75)
	at com.ilxqx.dubbo.demo.HelloServiceDubboProxy1.sayHello(HelloServiceDubboProxy1.java)
	at com.ilxqx.dubbo.demo.Consumer.main(Consumer.java:32)
Caused by: java.util.concurrent.TimeoutException: Timeout after 1000ms waiting for result.
	at org.apache.dubbo.rpc.AsyncRpcResult.get(AsyncRpcResult.java:207)
Caused by: java.util.concurrent.TimeoutException: Timeout after 1000ms waiting for result.

	at org.apache.dubbo.rpc.protocol.AbstractInvoker.waitForResultIfSync(AbstractInvoker.java:287)
	at org.apache.dubbo.rpc.protocol.AbstractInvoker.invoke(AbstractInvoker.java:190)
	at org.apache.dubbo.rpc.listener.ListenerInvokerWrapper.invoke(ListenerInvokerWrapper.java:71)
	at org.apache.dubbo.rpc.filter.RpcExceptionFilter.invoke(RpcExceptionFilter.java:41)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CopyOfFilterChainNode.invoke(FilterChainBuilder.java:334)
	at org.apache.dubbo.rpc.cluster.filter.FilterChainBuilder$CallbackRegistrationInvoker.invoke(FilterChainBuilder.java:196)
	at org.apache.dubbo.rpc.protocol.ReferenceCountInvokerWrapper.invoke(ReferenceCountInvokerWrapper.java:88)
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invokeWithContext(AbstractClusterInvoker.java:383)
	at org.apache.dubbo.rpc.cluster.support.FailoverClusterInvoker.doInvoke(FailoverClusterInvoker.java:80)
	... 27 more
```
<img width="1517" alt="image" src="https://github.com/fury-project/dubbo-serialization-fury/assets/18739624/47ab02bc-495a-45d6-9ed0-c4b59dfea3f8">

But when I use `fastjson2` or other impl can work:
<img width="1587" alt="image" src="https://github.com/fury-project/dubbo-serialization-fury/assets/18739624/8878e908-916a-40ef-8ad0-3de244e94130">

**Comment by chaokunyang** [comment]
@ilxqx Thanks for report the bug, I'll take a loot at it today.

**Comment by chaokunyang** [comment]
Hi @ilxqx , I fixed it in #3, tested in https://github.com/chaokunyang/fury-dubbo-example, it works. Could you test it again?

**Comment by chaokunyang** [comment]
Hi @ilxqx ,  I've fixed this in #3 and #5, could you test it again? I tested in https://github.com/chaokunyang/fury-dubbo-example, it works as expected.

If it still doesn't work, please reopen this issue or open a new issue

**Comment by coldsmirk** [comment]
@chaokunyang Yes, it works now. Thanks !
